Document registration update behavior (#170)
rancher/elemental-docs: docs/inventory-management.md

### MachineInventory

The `MachineInventory` holds all the relevant information for a registered machine.  
Upon successful registration, the `MachineInventory` will inherit all `machineInventoryLabels` defined in the related `MachineRegistration`.  
Additionally, the machine `annotations` will also be updated on each successful registration.  

> By default, Elemental Teal machines will attempt a registration update every 24 hours to update labels and annotations.  

Author: It does attempt registration at boot, but most likely will be skipped due to the 24 hours timer on the last successful updated. 
I omitted it to not mislead users who will otherwise expect an update every boot.
I think we can address it here: https://github.com/rancher/elemental-operator/issues/481

    # the MachineRegister approach
    machineUUID: f266c64b-3972-40e7-9937-3dc4a311436c
    manufacturer: QEMU
    productName: Standard-PC-Q35-ICH9-2009
    serialNumber: Not-Specified
    # Custom labels can be applied to each MachineInventory
    myCustomLabel: foo 
  name: m-479ab68e-00ff-4081-a731-5b1a76610289
  # The namespace must match the namespace of the cluster
  # assigned to the clusters.provisioning.cattle.io resource
  namespace: fleet-default
  # A reference to the MachineInventorySelector that links the 
  # machine to a Cluster definition
  ownerReferences:
  - apiVersion: elemental.cattle.io/v1beta1
    controller: true
    kind: MachineInventorySelector
    name: fire-machine-selector-qcn7d
    uid: 0a1f751e-4ca9-4a0d-919a-97ba1f434d12
spec:
  # The hash of the TPM EK public key. This is used if you are
  # using TPM2 to identifiy nodes.  You can obtain the TPM by
  # running `rancherd get-tpm-hash` on the node. Or nodes can
  # report their TPM hash by using the MachineRegister
  tpmHash: d68795c6192af9922692f050b...
```

### MachineRegistration

`MachineRegistration` holds information on how to install and configure all connected Elemental Teal machines.  

> It is possible to update the `spec.machineInventoryLabels` and `spec.machineInventoryAnnotations`, so that they will be applied to all registered machines.  
> By default, Elemental Teal machines will attempt a registration update every 24 hours to update labels and annotations.  

Reviewer: ```suggestion
It's possible to update the `spec.machineInventoryLabels` and `spec.machineInventoryAnnotations` and this will be applied to all registered machines.
By default, Elemental Teal machines will attempt a registration update every 24 hours to update labels and annotations.
```

> While it is possible to modify the `spec.config` definition, updates to the `spec.config` will be ignored by machines that already completed installation.  
> Machines that could not complete installation will try again every 30 minutes by default, reloading the remote `MachineRegistration` definition. This can be useful to correct `spec.config` mistakes that prevent successful installation (for ex. `spec.config.elemental.install.device`), without having to create a new `MachineRegistration` and a new ISO.  

Reviewer: ```suggestion
While it's possible to modify the `spec.config` definition, updates to the `spec.config` will be ignored by machines that already completed installation.
Machines that couldn't complete the installation will try again every 30 minutes by reloading the remote `MachineRegistration` definition. This can be useful to correct `spec.config` mistakes that prevent successful installation (for ex. `spec.config.elemental.install.device`), without having to create a new `MachineRegistration` and a new ISO.  
```

Reviewer: An admonition (info) could be used here to describe the reloading time as default, and also inform where and how to change it.
